internal/featuregates/features.go

```diff
@@ -45,3 +46,16 @@ func GetFeatureGateConfigMapName(feature string) string {
 	return strings.ToLower(feature) + "-config"
 
 }
+
+// Check if the configmap is a feature gate configmap
+// We use explicit check for the feature gate configmap
+// and avoid reconstructing the feature name from configmap name
+// and do the match
+func IsFeatureGateConfigMap(configMapName string) bool {
+	switch configMapName {
+	case FeatureGatesConfigMapName, LayeredImageDeploymentConfig:
```

Comment:
It could be worth the effort to find a more dynamic approach here.  Otherwise we're risking bugs due to forgetting to update this function, I'm afraid.

Reply:
I would prefer to keep it static here and document it in the code. It's possible to reconstruct by checking if configmap has the suffix `-config` and then string match the prefix with the featuregates. It's documented in the code as well so at least a minimal expectation from a new implementation ;)